Sources/Fullscreen/SearchResultMapView/SearchResultMapView.swift

```diff
@@ -0,0 +1,186 @@
+        let region = MKCoordinateRegion(center: coordinate, latitudinalMeters: regionDistance, longitudinalMeters: regionDistance)
+        mapView.setRegion(region, animated: animated)
+    }
+
+    public func setCenterOnUserLocation(regionDistance: CLLocationDegrees, animated: Bool) {
+        if mapView.userLocation.location != nil {
+            setCenter(mapView.userLocation.coordinate, regionDistance: regionDistance, animated: animated)
+        }
+    }
+
+    public func setMapOverlay(_ newOverlay: MKTileOverlay) {
+        if let lastOverlay = mapView.overlays.last {
+            mapView.removeOverlay(lastOverlay)
+        }
+
+        mapView.addOverlay(newOverlay)
+    }
+
+    public func clearMapOverlay(_ overlay: MKTileOverlay) {
+        mapView.removeOverlay(overlay)
+    }
+
+    public func addAnnotation(_ annotation: SearchResultMapViewAnnotation) {
+        mapView.addAnnotation(annotation)
+    }
+
+    public func clearAnnotations() {
+        mapView.removeAnnotations(mapView.annotations)
+    }
+
+    public func focusAnnotations() {
+        let annotationsToShow = mapView.annotations.filter { $0 is SearchResultMapViewAnnotation }
+        mapView.showAnnotations(annotationsToShow, animated: true)
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        backgroundColor = .milk
+
+        addSubview(mapView)
+
+        mapView.fillInSuperview()
+        mapView.addSubview(mapSettingsButton)
+
+        NSLayoutConstraint.activate([
+            mapSettingsButton.topAnchor.constraint(equalTo: mapView.topAnchor, constant: 8),
+            mapSettingsButton.trailingAnchor.constraint(equalTo: mapView.trailingAnchor, constant: -8)
+            ])
+    }
+
+}
+
+// MARK: - Extensions
+
+extension SearchResultMapView: MapSettingsButtonDelegate {
+
+    public func mapSettingsButtonDidSelectChangeMapTypeButton(_ view: MapSettingsButton) {
+        delegate?.searchResultMapViewDidSelectChangeMapTypeButton(self, mapSettingsButtonView: view)
+    }
+
+    public func mapSettingsButtonDidSelectCenterMapButton(_ view: MapSettingsButton) {
+        delegate?.searchResultMapViewDidSelectCenterMapButton(self)
+    }
+
+}
+
+extension SearchResultMapView: MKMapViewDelegate {
+
+    public func mapView(_ mapView: MKMapView, rendererFor overlay: MKOverlay) -> MKOverlayRenderer {
+        if let overlay = overlay as? MKTileOverlay {
+            return MKTileOverlayRenderer(tileOverlay: overlay)
+        }
+
+        return MKOverlayRenderer(overlay: overlay)
+    }
+
+    public func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        guard !(annotation is MKUserLocation) else { return nil }
+
+        if let annotation = annotation as? SearchResultMapViewAnnotation {
+            let marker = MKAnnotationView(annotation: annotation, reuseIdentifier: annotation.isCluster ? "clusterPOI" : "POI")
```

Comment:
There's something strange about having the annotations reuse identifier as hardcoded strings here, maybe we can make a String enum or something safely typed?

Reply:
Absolutely agree! I'll extract them.